tests/utils_image.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+
+# image related utilities
+
+import PIL
+import numpy as np
+
+
+def assert_pixel_by_pixel_equal(im1, im2):
+    """
+    Check that two PIL images are equal
+    pixel-by-pixel.
+    """
+    # see https://stackoverflow.com/questions/1927660/compare-two-images-the-python-linux-way
+    # compute pixel-by-pixel difference
+    diff = PIL.ImageChops.difference(im1, im2)
+    # if no difference, array is all 0's
+    diffa = np.array(diff)
+    assert np.sum(diffa) == 0
```

Comment:
What do you think about changing it to 
```suggestion
    assert np.count_nonzero(diffa) == 0
```

While the current code is correct, it is not straightforward: `sum(X)` can be 0 if some elements are positive and some are negative, so this check ("array is all 0's") only works if X is non-negative, and it is far from obvious that ImageChops.difference returns abs(difference), not just a difference.

Reply:
I agree. This is a good change. I have used `np.count_nonzero` in another test to check for all 0's and it makes sense.